plugins/modules/ndo_l3out_node_routing_policy.py

```diff
@@ -0,0 +1,561 @@
+        template=dict(type="str", required=True),
+        name=dict(type="str", aliases=["l3out_node_routing_policy_name"]),
+        uuid=dict(type="str", aliases=["l3out_node_routing_policy_uuid"]),
+        description=dict(type="str"),
+        bfd_multi_hop_settings=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                admin_state=dict(type="str", choices=["enabled", "disabled"]),
+                detection_multiplier=dict(type="int"),
+                min_receive_interval=dict(type="int"),  # msec
+                min_transmit_interval=dict(type="int"),  # msec
+            ),
+        ),
+        bgp_node_settings=dict(
+            type="dict",
+            options=dict(
+                state=dict(type="str", choices=["enabled", "disabled"]),
+                graceful_restart_helper=dict(type="str", choices=["enabled", "disabled"]),
+                keep_alive_interval=dict(type="int"),  # sec
+                hold_interval=dict(type="int"),  # sec
+                stale_interval=dict(type="int"),  # sec
+                max_as_limit=dict(type="int"),
+            ),
+        ),
+        as_path_multipath_relax=(dict(type="str", choices=["enabled", "disabled", ""])),
+        state=dict(type="str", default="query", choices=["absent", "query", "present"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["name", "uuid"], True],
+            ["state", "present", ["name", "uuid"], True],
+        ],
+    )
+
+    mso = MSOModule(module)
+
+    template = module.params.get("template")
+    name = module.params.get("name")
+    uuid = module.params.get("uuid")
+    description = module.params.get("description")
+    bfd_multi_hop_settings = module.params.get("bfd_multi_hop_settings")
+    bgp_node_settings = module.params.get("bgp_node_settings")
+    as_path_multipath_relax = module.params.get("as_path_multipath_relax")
+    state = module.params.get("state")
+
+    if bfd_multi_hop_settings is not None:
+        check_int_between(mso, "bfd_multi_hop_settings.detection_multiplier", bfd_multi_hop_settings.get("detection_multiplier"), 1, 50)
+        check_int_between(mso, "bfd_multi_hop_settings.min_receive_interval", bfd_multi_hop_settings.get("min_receive_interval"), 250, 999)
+        check_int_between(mso, "bfd_multi_hop_settings.min_transmit_interval", bfd_multi_hop_settings.get("min_transmit_interval"), 250, 999)
+
+    if bgp_node_settings is not None:
+        check_int_between(mso, "bgp_node_settings.keep_alive_interval", bgp_node_settings.get("keep_alive_interval"), 0, 3600)
+        check_int_between(mso, "bgp_node_settings.stale_interval", bgp_node_settings.get("stale_interval"), 1, 3600)
+        check_int_between(mso, "bgp_node_settings.max_as_limit", bgp_node_settings.get("max_as_limit"), 0, 2000)
+
```

Comment:
Doesn't API return an error when the value is out of range? if API returns an error, then we do not require check_int_between() function